### ISSUE TITLE
allow dependency versioning requirements

### ISSUE THREAD

**Issue by binaryphile** [basherpm/basher#20]
The DEPS specification allows me to pull particular packages from github.  However, I would also like to be able to specify the particular version of the desired package, since master is usually a moving target and can break the contract with my code over time.

The simplest thing would be to specify a tag like "v1.0" which would check out that tag from the repository after it had been cloned.

This would also affect the upgrade command, since this package wouldn't be updated without changing the version specified in DEPS.

I'm not requesting that multiple versions of the dependency be able to coexist under basher, although that would be another nice-to-have (i.e. different feature request).  I could definitely see specifying the version number in the repo's path in anticipation of such a feature, however, such as "cellar/packages/my-user/my-package-v0.1".

**Comment by juanibiapina** [comment]
I agree this is a great feature, not only for deps, but also for normal installs.

**Comment by juanibiapina** [comment]
@binaryphile how do you like the work on #27 ? I haven't tested it with dependencies, but it seems like it should work.
